Update column visibility at the end of rendering the treeview [PREVIEW]

nicoe opened the pull request:
For performance reasons it's better not to mix DOM reading and modifying especially in JS loops.

PCLAS-772

jmousset reviewed (commented):
par contre, si ça reste en spécifque, pour faciliter les merges futures, il faudra ajouter des commentaire de type "Coog Specific: see #xxxx"

nicoe commented:
> par contre, si ça reste en spécifque, pour faciliter les merges futures, il faudra ajouter des commentaire de type "Coog Specific: see #xxxx"

OK mais quand c'est un changement qui touche quelques endroits, vous faites comment ? Juste sur un bout du patch ou sur tous les bouts ?

jmousset commented:
@nicoe il faudra qu'on trouve une manière plus élégante de faire, mais là je dirais rajouter un ligne de commentaire sur tous les morceaux de diff.

nicoe committed:
Update column visibility at the end of rendering the treeview [PREVIEW]

For performance reasons it's better not to mix DOM reading and
modifying especially in JS loops.

PCLAS-772

JCavallo commented:
LGTM

havaguernicoopengo commented on sao/src/view/tree.js:
```diff
@@ -985,6 +984,38 @@
+            }
+            if (to_show.length) {
+                this.tbody.find(to_show.map(make_selector).join(','))
+                    .removeClass('invisible').show();
```
Petit problème ici avec jQuery qui repasse un coup de recalculate style sur tout le DOM en itérant sur le body j'ai l'impression. J'ai cherché des solutions pour éviter ça mais no luck.

nicoe replied:
Je pensais naïvement qu'en utilisant un selecteur ça ne ferait qu'une seule invalidation.
